include/seqan3/core/tuple_utility.hpp

```diff
@@ -0,0 +1,184 @@
+ * Note, that the returned tuples can be empty. For this reason it is not possible to use tuple like objects,
+ * that cannot be empty, i.e. std::pair. Using such an object will emit an compiler error.
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements.
+ *
+ * ### Thread safety
+ *
+ * Concurrent invocations of this functions are thread safe.
+ */
+template <size_t pivot_c, template <typename ...> typename tuple_t, typename ... ts>
+    requires tuple_like_concept<tuple_t<ts...>>
+constexpr auto tuple_split(tuple_t<ts...> const & t)
+{
+    static_assert(pivot_c < sizeof...(ts));
+
+    return tuple_t{detail::tuple_split<0>(t, std::make_index_sequence<pivot_c>{}),
+                   detail::tuple_split<pivot_c>(t, std::make_index_sequence<sizeof...(ts) - pivot_c>{})};
+}
+
+//!\copydoc seqan3::tuple_split
+template <size_t pivot_c, template <typename ...> typename tuple_t, typename ... ts>
+    requires tuple_like_concept<tuple_t<ts...>>
+constexpr auto tuple_split(tuple_t<ts...> && t)
+{
+    static_assert(pivot_c < sizeof...(ts));
+
+    return tuple_t{detail::tuple_split<0>(std::move(t), std::make_index_sequence<pivot_c>{}),
+                   detail::tuple_split<pivot_c>(std::move(t), std::make_index_sequence<sizeof...(ts) - pivot_c>{})};
+}
+
+/*!\brief Splits a tuple like data structure at the first position of the given type.
+ * \ingroup core
+ *
+ * \tparam    pivot_t A template type specifying the split position.
+ * \param[in] t       The original tuple to split.
+ *
+ * \returns A new tuple of tuples with the left side of the split and the right side of the split.
+ *
+ * \copydetails seqan3::tuple_split
+ */
+template <typename pivot_t, tuple_like_concept tuple_t>
+constexpr auto tuple_split(tuple_t && t)
+{
+    using as_list_t = detail::transfer_template_args_onto_t<remove_cvref_t<tuple_t>, type_list>;
+    constexpr size_t pivot_c = meta::find_index<as_list_t, pivot_t>::value;
+
+    static_assert(pivot_c <= std::tuple_size_v<remove_cvref_t<tuple_t>>);
+
+    return tuple_split<pivot_c>(std::forward<tuple_t>(t));
+}
+
+/*!\brief Removes the first element of a tuple.
+ * \ingroup core
+ *
+ * \param[in] t  The original tuple.
+ *
+ * \returns A new tuple without the first element of `t`.
+ *
+ * \details
+ *
+ * Note, that the tuple must contain at least one element and must support empty tuple types, i.e. std::pair cannot
+ * be used.
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements.
+ *
+ * ### Thread safety
+ *
+ * Concurrent invocations of this functions are thread safe.
+ */
+template <tuple_like_concept tuple_t>
+constexpr auto tuple_pop_front(tuple_t && t)
+{
+    static_assert(std::tuple_size_v<remove_cvref_t<tuple_t>> > 0);
+
+    if constexpr (std::tuple_size_v<remove_cvref_t<tuple_t>> == 1)
+        return std::get<0>(tuple_split<0>(std::forward<tuple_t>(t)));
+    else
+        return std::get<1>(tuple_split<1>(std::forward<tuple_t>(t)));
```

Comment:
assuming `[t_0, ..., t_{p-1}], [t_p, ..., t_{n-1}]`:
```c++
tuple_split<0>([t_0]) = [ ], [t_0]
std::get<0>(tuple_split<0>([t_0])) = std::get<0>([ ], [t_0]) = [ ]
```

but

```c++
tuple_split<1>([t_0]) = [t_0], [ ]
std::get<1>(tuple_split<1>([t_0])) = std::get<1>([t_0], [ ]) = [ ]
```
Has the same behaviour as the general case.